sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetricInfo.scala

```diff
@@ -27,4 +27,5 @@ import org.apache.spark.annotation.DeveloperApi
 class SQLMetricInfo(
     val name: String,
     val accumulatorId: Long,
-    val metricType: String)
+    val metricType: String,
+    val stats: Long = -1)
```

Comment:
not all metric has its corresponding statistics (e.g. `peakMemory`), and not all statistics are long type. We should think of a better place to carry the statistics.

Reply:
or we can put a `val stats: Option[Statistics] = None`

Reply:
@cloud-fan Thanks for your comments.

Put `Option[Statistics]` in `SQLMetricInfo` doesn't sound quite right though. It means that all SQL metrics have an attribute including rowCount, size & column stats. 

`SQLMetric(val metricType: String, initValue: Long = 0L, val stats: Long = -1L)`
SQLMetric's value is `Long` type, it makes sense to have a stats with Long type as well, doesn't it?

Let me know your feedback, thanks in advance.

Reply:
@cloud-fan Thanks for your comments.

The idea is that each SQL metric can have a statistic value (-1 means not available/initialized). I set the statistic type to Long is because SQL Metric's value is always Long type as well. `class SQLMetric(val metricType: String, initValue: Long = 0L)`

Put Option[Statistics] in SQLMetricInfo doesn't sound quite right though. It means that all SQL metrics have an attribute including rowCount, size & column stats.

Let me know your feedback, thanks in advance.